**Ms. Anita L. Devkar**

Please find Exercise Solution Below

## Exercises

Answer the questions or complete the tasks outlined in bold below, use the specific method described if applicable.

** What is 7 to the power of 4?**

In [ ]:
7**4
print(7**4)

2401


** Split this string:**

    s = "Hi there Yogesh!"
    
**into a list. **

In [ ]:
s = "Hi there Yogesh!"      

In [ ]:
x=s.split(' ')                    #split() is used for splitting 
print(x)

['Hi', 'there', 'Yogesh!']


In [ ]:
s.split()             #another solution

['Hi', 'there', 'Yogesh!']

** Given the variables:**

    planet = "Earth"
    diameter = 12742

** Use .format() to print the following string: **

    The diameter of Earth is 12742 kilometers.

In [ ]:
planet = "Earth"
diameter = 12742
print('The diameter of {} is {} kilometers.'.format(planet,diameter))

The diameter of Earth is 12742 kilometers.


** Given this nested list, use indexing to grab the word "hello" **

In [ ]:
lst = [1,2,[3,4],[5,[100,200,['hello']],23,11],1,7]

In [ ]:
lst[3][1][2]

['hello']

In [ ]:
print(lst[3][1][2][0])

hello


** Given this nested dictionary grab the word "hello". Be prepared, this will be annoying/tricky **

In [ ]:
d = {'k1':[1,2,3,{'tricky':['oh','man','inception',{'target':[1,2,3,'hello']}]}]}

In [ ]:
d['k1'][3]['tricky'][3]['target'][3]

'hello'

** What is the main difference between a tuple and a list? **

In [ ]:
# Tuples are immutable while lists are mutable
#Tuples are more memory efficient than the lists.
#when we considered time efficiency factor so again tuple is better than list

** Create a function that grabs the email website domain from a string in the form: **

    user@domain.com
    
**So for example, passing "user@domain.com" would return: domain.com**

In [ ]:
def getdomainfunc(email):                       #function defination
    return email.split('@')[1]

In [ ]:
getdomainfunc('user@domain.com')       #Function calling

'domain.com'

In [ ]:
getdomainfunc('anitadevkar@gmail.com')

'gmail.com'

** Create a basic function that returns True if the word 'dog' is contained in the input string. Don't worry about edge cases like a punctuation being attached to the word dog, but do account for capitalization. **

In [ ]:
def searchdog(input):
  return 'dog' in input.lower()

In [ ]:
searchdog('Is there a dog here?')

True

** Create a function that counts the number of times the word "dog" occurs in a string. Again ignore edge cases. **

In [ ]:
def countDog(input):
    count = 0
    for i in input.lower().split():
        if i == 'dog':
            count += 1
    return count


In [ ]:
countDog('This dog runs faster than the other dog dude!')

2

** Use lambda expressions and the filter() function to filter out words from a list that don't start with the letter 's'. For example:**

    seq = ['soup','dog','salad','cat','great']

**should be filtered down to:**

    ['soup','salad']

In [ ]:
seq = ['soup','dog','salad','cat','great']

In [ ]:
list(filter(lambda x: x[0]=='s',seq))

['soup', 'salad']

### Final Problem
**You are driving a little too fast, and a police officer stops you. Write a function
  to return one of 3 possible results: "No ticket", "Small ticket", or "Big Ticket". 
  If your speed is 60 or less, the result is "No Ticket". If speed is between 61 
  and 80 inclusive, the result is "Small Ticket". If speed is 81 or more, the result is "Big    Ticket". Unless it is your birthday (encoded as a boolean value in the parameters of the function) -- on your birthday, your speed can be 5 higher in all 
  cases. **

In [ ]:
def caught_speeding(speed, is_birthday):
    
    if is_birthday:
        spiding = speed - 5
    else:
        spiding = speed
    
    if spiding > 80:
        return 'Big Ticket'
    elif spiding > 60:
        return 'Small Ticket'
    else:
        return 'No Ticket'

In [ ]:
caught_speeding(81,True)

'Small Ticket'

In [ ]:
caught_speeding(81,False)

'Big Ticket'

In [ ]:
Question
Given a string print number num by extracting all the
digits from the string as in string
Print negative number if the first character in 
string is '-'
eg.
-123abcd should return -123,abcd456-->456,fdhfh78dsd89--->7889,-56dssd78-->-5678

In [70]:
input_string =input('Enter the string')
temp_str = ""
for m in input_string:
    if m.isdigit():
        temp_str = temp_str + m
print("Find numbers from string:",temp_str) 


Enter the string-123abcd
Find numbers from string: 123


In [ ]:
Q2
WAP to check whether a number is palindrome or not on following conditions.
1. take input number
2. add number and its reverse
3. check that number is palindrome or not,if not then sum and its reverse

In [ ]:
n=int(input("Enter number:"))
temp=n
rev=0
sum=0
while(n>0):
    dig=n%10
    rev=rev*10+dig
    sum=sum+dig
    n=n//10
if(temp==rev):
    print("The number is a palindrome!")
else:
    print('Number reverse is {one} and number sum is {two}'.format(one=rev,two=sum))

   


Enter number:345
Number reverse is 543 and number sum is 12


**Output1:**

Enter number:121

The number is a palindrome!

**Output2:** 

Enter number:345

Number reverse is 543 and number sum is 12

In [ ]:
Write a python function to find and display the five 
digit number in which the first digit is two more than 
the second,the second digit is two more than the third,
the fourth digit is two less than the third, and the 
last digit is two more than the fourth.The sum of the third,fourth and fifth digits
equals the first.The sum of all the digits is 19

In [ ]:
def find_five_digit():
   for i in range(10,0,-1):
       first=i
       second = i-2
       third=i-4
       fourth=i-6
       fifth=i-4   
       if third+fourth+fifth==first:
           if (first+second+third+fourth+fifth)==19:
               return(int(str(first)+str(second)+str(third)+str(fourth)+str(fifth)))
print(find_five_digit())   
#print(7+5+3+1+3) 

75313
